### ISSUE TITLE
Request for data in CSV file

### ISSUE THREAD

**Issue by ZamirKhurshid** [alirezaniki/OBSTransformer#1]
Hello sir hope you are well. I have downloaded data but dataset is present in to files named as 01 & 02, In which data is present in miniseed format. Do we need to change this data into CSV format one by one and then merege in one file. Can we use miniseed file directly in to python. Your response related to querry will be encourging.
Thanks

**Comment by alirezaniki** [comment]
@ZamirKhurshid The dataset in this example has the following structure:
- dataset |
           -- station 1 |
                    --- miniseed files
           -- station 2 |
                    --- miniseed files        
           -- station 3 |
                    --- miniseed files                                       

where each miniseed file needs to be stored in the following convention:
network.station.location.channel__starttime__endtime.mseed

To apply the pre-trained OBSTransformer to your data, you don't need a CSV file. Just prepare the dataset and use the code provided here (detection.py) to do the detection and picking.
The CSV and HdF5 files are required for fine-tuning. The HfF5 file includes the seismograms and the CSV file acts as the metadata. See EQTransfofmer (https://github.com/smousavi05/EQTransformer) and STEAD (https://github.com/smousavi05/STEAD) GitHub pages for more details.

**Comment by ZamirKhurshid** [comment]
Thanks for your response sir. Can you explain a little bit more how we prepare a single miniseed file(how we prepare data) because in the data set 1 & 2 there are many miniseed files.

**Comment by alirezaniki** [comment]
@ZamirKhurshid In the example dataset, there are three 3-component seismograms each ~1 minute long (pay attention to start time and end times). In general,  each miniseed data in the dataset needs to be at least 60 seconds long (6000 samples). It means that you can have your data in any length (1 hour, 1 day, 1 month, etc.) but not shorter than 60 sec. 
Let's say you want to apply OBST on 1 day of data from network BB, station AA, channel BH; then you can store the data as below (one-day data in one file):
```
AA.BB..BHZ__20230703T000000Z__20230704T000000Z.mseed
AA.BB..BH1__20230703T000000Z__20230704T000000Z.mseed
AA.BB..BH2__20230703T000000Z__20230704T000000Z.mseed
```

**Comment by ZamirKhurshid** [comment]
Thank you sir for your prompt response. I am trying to implement my code
but it shows errors. I humbly request you please share code with me so that
I can solve my problem. I will be very thankful to you sir.

**Comment by alirezaniki** [comment]
@ZamirKhurshid send me the errors here so that I can assist you

**Comment by ZamirKhurshid** [comment]
@alirezaniki Downloading data from IRIS
Sir how we download data from IRIS, When we give UTCDateTime it shows there is no data available. Please have a look on screen as well. What UTCDateTime give so that data will be downloded from IRIS.
![Python file](https://github.com/alirezaniki/OBSTransformer/assets/108117171/d3e4dd0b-470d-4da1-ba95-511ea0a6a507)

**Comment by alirezaniki** [comment]
@ZamirKhurshid  This error is not directly related to OBSTransformer and should be asked in the Obspy forum. 
You should run all blocks from top to bottom subsequently. The error says that UTCDateTime is not imported.

**Comment by ZamirKhurshid** [comment]
@alirezaniki Yes I know this error is not related to OBSTransformer, I have run data on obspy and get  seismic wave forms.
Now how we download earthquake catalouge accross these waveforms from IRIS. I have tried but it shows no data is available. Also what origion UTCDateTime we used so that data will be downloded accross these waveforms from IRIS.

**Comment by alirezaniki** [comment]
Check out the below link:
https://github.com/Dal-mzhang/LOC-FLOW
On folder Data/ you can find scripts to download waveforms from different providers along with their metadata.

**Comment by ZamirKhurshid** [comment]
@alirezaniki I have explore the link and have used that program as well but it shows no data is available.
Can you please share direct IRIS link from where I can download earthquake catalouge data set.

**Comment by ZamirKhurshid** [comment]
I want to download meta data related to earthquake. I want to download the
data across the station. Which are given in your file on github.  Named
dataset 1 and 2.I want to download data across these wave form station.

**Comment by alirezaniki** [comment]
read below link:
https://docs.obspy.org/packages/autogen/obspy.clients.fdsn.mass_downloader.html

As provided, you need to change the variables (time, reference location, channels, stations, network, etc) correctly to get the correct metadata:
```
import obspy
from obspy.clients.fdsn.mass_downloader import CircularDomain, \
    Restrictions, MassDownloader


domain = CircularDomain(latitude=23, longitude=121.5,
                        minradius=0, maxradius=10)

restrictions = Restrictions(
    # Get data for a whole year.
    starttime=obspy.UTCDateTime(2008, 11, 23, 0, 49, 14),
    endtime=obspy.UTCDateTime(2008, 11, 23, 0, 55, 14),
    # Chunk it to have one file per day.
    chunklength_in_sec=86400,
    # Considering the enormous amount of data associated with continuous
    # requests, you might want to limit the data based on SEED identifiers.
    # If the location code is specified, the location priority list is not
    # used; the same is true for the channel argument and priority list.
    network="YM", station="01,02", location="", channel="BH*",
    # The typical use case for such a data set are noise correlations where
    # gaps are dealt with at a later stage.
    reject_channels_with_gaps=False,
    # Same is true with the minimum length. All data might be useful.
    minimum_length=0.0,
    # Guard against the same station having different names.
    minimum_interstation_distance_in_m=100.0)

# Restrict the number of providers if you know which serve the desired
# data. If in doubt just don't specify - then all providers will be
# queried.
mdl = MassDownloader(providers=["IRIS"])
mdl.download(domain, restrictions, mseed_storage="waveforms",
             stationxml_storage="stations")

```

Here the data and metadata will be stored in the waveforms/ and stations/ folders in .mseed and .XML format. You can read them:

```
from obspy import read, read_inventory
data = read('waveforms/*mseed')
metadata = read_inventory('stations/*xml')
for trace in data:
    print(metadata.get_channel_metadata(trace.id))
```

Hope this helps.

**Comment by ZamirKhurshid** [comment]
@alirezaniki Hello bro thanks for your help. I have set parameters according to data set 01,02 but it shows error please have a look on this.

**Comment by ZamirKhurshid** [comment]
Hello bro thanks for your help. I have set parameters according to data set 01,02 but it shows error please have a look on this.
![ZEE file](https://github.com/alirezaniki/OBSTransformer/assets/108117171/ce12ce66-0e09-44f8-9bda-83df502082b2)

**Comment by alirezaniki** [comment]
The last line needs to end with a parenthesis! 
```
mdl.download(domain, restrictions, mseed_storage="waveforms",
             stationxml_storage="stations")
```

**Comment by ZamirKhurshid** [comment]
@alirezaniki I have checked with that as well please check your email.

**Comment by ZamirKhurshid** [comment]
Aoa bro hope you are well. Please share data or direct link of data so that
I can download it. I am trying but I am unable to download it. Jazakullah


Kind Regards:

**Comment by alirezaniki** [comment]
Hi,

I closed the issue on GitHub. In your last post, I couldn't find any codes! Please note that GitHub is not a chatroom, and you should be precise and concise.
I advise you to start learning Obspy. Meanwhile, send me the code here, and I'll take a look at it.


#~~~~~~~~~~~~~
Alireza Niksejel
Ph.D. Candidate
Earth and Environmental Sciences
Life Sciences Center, Dalhousie University
Email: ***@***.***
[cid:72da00ed-7389-4eda-88a3-f7253b9c3c89]<http://www.linkedin.com/in/alireza-niksejel>
________________________________

**Comment by ZamirKhurshid** [comment]
Well noted thank you.

**Comment by ZamirKhurshid** [comment]
Well noted thank you sir for your response and suggestions. I am
learning Obspy but I am a little bit confused about data. I have tried many
times to download the data but was not successful. I am sending you my code
again please have a look at it, where I am making a mistake. Jazakallah

**Comment by ZamirKhurshid** [comment]
Thanks brother for your response and guidance May ALLAH and SAW bless you
and give  you more progress. I have attached code with the mail. I am
sending you again. My supervisor told me that you have to work on the OBS
dataset. My research topic is phase picking and event detection through OBS
data set. For this I need an OBS data set. Sorry for inconvenience to you.
If you help me in this regard related to data I will be thankful to you and
will remember your help forever.
Please have a look at my code.
Jazakallah
